src/main/java/analysis/GoblintAnalysis.java

```diff
@@ -76,14 +81,20 @@ public void analyze(Collection<? extends Module> files, AnalysisConsumer consume
         if (rerun) {
             if (consumer instanceof MagpieServer server) {
 
+                if (analysisRunning) { 
+                    abortAnalysis();
+                }
                 goblintConfObserver.checkAndNotify();
                 preAnalyse();
 
                 System.err.println("\n---------------------- Analysis started ----------------------");
+                analysisRunning = true;
                 Collection<GoblintAnalysisResult> response = reanalyse();
-                if (response != null) server.consume(new ArrayList<>(response), source());
-                System.err.println("--------------------- Analysis finished ----------------------\n");
-
+                if (response != null) {
+                    server.consume(new ArrayList<>(response), source());
+                    analysisRunning = false;
+                    System.err.println("--------------------- Analysis finished ----------------------\n");
```

Comment:
A more general side point to maybe address separately: this `println`s might be better replaced with `log.info`.